pkg/vf/virtio.go

```diff
@@ -289,6 +290,73 @@ func (dev *VirtioVsock) AddToVirtualMachineConfig(vmConfig *VirtualMachineConfig
 	return nil
 }
 
+func (dev *NetworkBlockDevice) toVz() (vz.StorageDeviceConfiguration, error) {
+	if dev.Uri == "" || dev.DeviceIdentifier == "" {
+		return nil, fmt.Errorf("missing required options for NBD device: 'uri' and 'deviceId' must be specified")
```

Comment:
here you check for empty options, but do you ever check to see if they are valid?  If you do and you confident in that, maybe move this to that section?